FinnUI/Sources/Components/SearchFilterTags/SearchFilterTagsView.swift

```diff
@@ -0,0 +1,255 @@
+        return view
+    }()
+
+    private let viewModel: SearchFilterTagsViewModel
+    private var searchFilterTags = [SearchFilterTagCellViewModel]()
+
+    private var maxFilterButtonWidth: CGFloat {
+        filterButtonView.contentWidth
+    }
+
+    private var minFilterButtonWidth: CGFloat {
+        SearchFilterButtonView.minWidth
+    }
+
+    private var filterButtonWidthConstraint: NSLayoutConstraint!
+
+    // MARK: - Init
+
+    public init(viewModel: SearchFilterTagsViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        backgroundColor = .bgPrimary
+
+        addSubview(filterButtonView)
+        addSubview(collectionView)
+        addSubview(separatorView)
+
+        filterButtonWidthConstraint = filterButtonView.widthAnchor.constraint(equalToConstant: maxFilterButtonWidth)
+
+        NSLayoutConstraint.activate([
+            filterButtonView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: SearchFilterTagsView.horizontalMargin),
+            filterButtonView.topAnchor.constraint(equalTo: topAnchor, constant: SearchFilterTagsView.verticalMargin),
+            filterButtonView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -SearchFilterTagsView.verticalMargin),
+            filterButtonWidthConstraint,
+
+            collectionView.leadingAnchor.constraint(equalTo: filterButtonView.trailingAnchor, constant: SearchFilterTagsView.filterButtonTrailingMargin),
+            collectionView.topAnchor.constraint(equalTo: topAnchor, constant: SearchFilterTagsView.verticalMargin),
+            collectionView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -SearchFilterTagsView.verticalMargin),
+            collectionView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            collectionView.heightAnchor.constraint(equalToConstant: SearchFilterTagCell.height),
+
+            separatorView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            separatorView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            separatorView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            separatorView.heightAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale),
+        ])
+    }
+
+    // MARK: - Public methods
+
+    public func reloadData() {
+        collectionView.reloadData()
+    }
+
+    public func configure(with searchFilterTags: [SearchFilterTagCellViewModel], reloadSection: Bool = true) {
+        let searchFilterTagsDidChange = !searchFilterTagsAreEqual(self.searchFilterTags, searchFilterTags)
+        self.searchFilterTags = searchFilterTags
+
+        guard
+            searchFilterTagsDidChange,
+            reloadSection
+        else { return }
+
+        UIView.performWithoutAnimation {
+            collectionView.reloadSections([0])
```

Comment:
I had a bit of trouble right here. Ideally I would just do a `reloadData()`. But I discovered that `reloadData()` does not reload all visible cells (I expected `cellForItemAt` to be called for all visible cells). It’s an issue in the rare cases where two search filter tags switch order, but the count of search filter tags did not change. Then the cells would still contain their old label value, since cells were not configured with the new data in `cellForItemAt`.

However, 
`reloadSections[0]`
or
```
collectionView.reloadItems(at: collectionView.indexPathsForVisibleItems)
collectionView.reloadData()
```
works, since they reload all visible cells.

Does anyone know why `reloadData()` doesn’t??

Reply:
Could it be because `reloadData()` is not called from the main thread?

Reply:
No, I checked and it is called from main thread 🤷‍♀️